eval/eval-scripts/mt_bench.py

```diff
@@ -0,0 +1,149 @@
+import requests
+
+def run_mt_bench(
+    model_s3_path, 
+    judge_model_name, 
+    server_url, 
+    output_dir="eval_output", 
+    output_s3_path=None, 
+    max_workers="auto", 
+    merge_system_user_message=False
+):
+    """
+    Runs the MT-Bench evaluation using the MTBenchEvaluator class.
+
+    Args:
+        model_s3_path (str): S3 path of the model to evaluate.
+        judge_model_name (str): Name or path of the judge model.
+        server_url (str): Model server endpoint (e.g., http://localhost:8000/v1).
+        output_dir (str): Directory for evaluation output.
+        output_s3_path (str): S3 path to upload the evaluation output.
+        max_workers (int | str): Max parallel workers (int or "auto").
+        merge_system_user_message (bool): Whether to merge system and user messages.
+    """
+
+    # Setup environment variables for AWS credentials
+    aws_access_key = os.getenv("AWS_ACCESS_KEY_ID", "your-access-key-id")
+    aws_secret_key = os.getenv("AWS_SECRET_ACCESS_KEY", "your-secret-access-key")
+    aws_region = os.getenv("AWS_REGION", "us-east-1")
+    s3_endpoint_url = os.getenv("S3_ENDPOINT_URL", "https://s3.amazonaws.com")
+    os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+
+    # Set up S3 client
+    s3_client = boto3.client(
+        "s3",
+        aws_access_key_id=aws_access_key,
+        aws_secret_access_key=aws_secret_key,
+        region_name=aws_region,
+        endpoint_url=s3_endpoint_url,
+    )
+
+    # OPENAI_API_KEY env var used to pass the token auth to create model server client
+    os.environ["OPENAI_API_KEY"] = os.getenv("OPENAI_API_KEY", "")
+
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    output_dir = f"{output_dir}_{timestamp}"
+
+    model_local_path = "/tmp/model_to_evaluate"
+    os.makedirs(model_local_path, exist_ok=True)
+
+    print(f"Attempting to download model from {model_s3_path} to {model_local_path}")
+
+    bucket_name = model_s3_path.split('/')[2]
+    prefix = '/'.join(model_s3_path.split('/')[3:])
+
+    # List and download model files
+    try:
+        response = s3_client.list_objects_v2(Bucket=bucket_name, Prefix=prefix)
+        if 'Contents' not in response:
+            print(f"No objects found at {model_s3_path}. Check the S3 path and permissions.")
+            return
+        
+        for obj in response['Contents']:
+            key = obj['Key']
+            local_file_path = os.path.join(model_local_path, os.path.basename(key))
+            s3_client.download_file(bucket_name, key, local_file_path)
+            print(f"Downloaded {key} to {local_file_path}")
+
+        if os.listdir(model_local_path):
+            print(f"Model files found at {model_local_path}.")
+        else:
+            print(f"Model files not found at {model_local_path}. Check the S3 path and permissions.")
+
+    except Exception as e:
+        print(f"Error downloading model from S3: {e}")
+        raise
```

Comment:
Same as above:

Can't you use here `dsl.importer` instead? It will be safer + less code to maintain + always using tracked artifacts.

https://www.kubeflow.org/docs/components/pipelines/user-guides/components/importer-component/

Maintaining 2 times 40 lines of S3 downloader code seems like a tech debt.

Reply:
why yes I can